File: app/src/main/java/com/commit451/gitlab/api/rss/DateTransform.java
package com.commit451.gitlab.api.rss;

import com.commit451.gitlab.util.ConversionUtil;

import org.simpleframework.xml.transform.Transform;

import java.util.Date;

/**
 * Transforms dates!
 */
public class DateTransform implements Transform<Date> {

    public DateTransform() {
    }

    @Override
    public Date read(String value) throws Exception {
        return ConversionUtil.toDate(value);
    }

    @Override
    public String write(Date value) throws Exception {
        return ConversionUtil.fromDate(value);
    }
}

File: app/src/main/java/com/commit451/gitlab/api/rss/DateTransform.kt
package com.commit451.gitlab.api.rss

import org.simpleframework.xml.transform.Transform
import org.threeten.bp.Instant
import org.threeten.bp.format.DateTimeFormatter
import java.util.*

/**
 * Transforms dates!
 */
class DateTransform : Transform<Date> {

    override fun read(value: String): Date? {
        return Date(Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(value)).toEpochMilli())
    }

    override fun write(value: Date): String {
        return DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(Instant.ofEpochMilli(value.time))
    }
}
